From c524695eda823d025f5d2f64b8a84e1b478caa9a Mon Sep 17 00:00:00 2001
From: Nicolas Bertrand <nicolas.bertrand@linux.intel.com>
Date: Thu, 4 Oct 2012 10:23:16 +0200
Subject: [PATCH 01/10] hal: Add set_sound_path function
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 include/hal.h |    2 ++
 src/hal.c     |   10 ++++++++++
 src/mux.c     |    8 ++++++++
 3 files changed, 20 insertions(+)

diff --git a/include/hal.h b/include/hal.h
index d93795a..55b4762 100644
--- a/include/hal.h
+++ b/include/hal.h
@@ -43,6 +43,7 @@ enum tcore_hal_mode {
 struct tcore_hal_operations {
 	TReturn (*power)(TcoreHal *hal, gboolean flag);
 	TReturn (*send)(TcoreHal *hal, unsigned int data_len, void *data);
+	TReturn (*set_sound_path)(TcoreHal *hal, int device);
 };
 
 TcoreHal*    tcore_hal_new(TcorePlugin *plugin, const char *name,
@@ -58,6 +59,7 @@ enum tcore_hal_mode tcore_hal_get_mode(TcoreHal *hal);
 TReturn      tcore_hal_set_mode(TcoreHal *hal, enum tcore_hal_mode mode);
 
 TReturn      tcore_hal_set_power(TcoreHal *hal, gboolean flag);
+TReturn	      tcore_hal_set_sound_path(TcoreHal *hal, int device);
 
 TReturn      tcore_hal_link_user_data(TcoreHal *hal, void *user_data);
 void*        tcore_hal_ref_user_data(TcoreHal *hal);
diff --git a/src/hal.c b/src/hal.c
index 178cecd..59ba41f 100644
--- a/src/hal.c
+++ b/src/hal.c
@@ -508,3 +508,13 @@ TReturn tcore_hal_set_power(TcoreHal *hal, gboolean flag)
 
 	return hal->ops->power(hal, flag);
 }
+
+TReturn tcore_hal_set_sound_path(TcoreHal *hal, int device)
+{
+	dbg("start");
+
+	if (!hal || !hal->ops || !hal->ops->set_sound_path)
+		return TCORE_RETURN_EINVAL;
+
+	return hal->ops->set_sound_path(hal, device);
+}
diff --git a/src/mux.c b/src/mux.c
index 46cafce..7456762 100644
--- a/src/mux.c
+++ b/src/mux.c
@@ -283,10 +283,18 @@ static TReturn tcore_cmux_hal_send(TcoreHal *h, unsigned int data_len, void *dat
 	return TCORE_RETURN_SUCCESS;
 }
 
+static TReturn tcore_cmux_hal_set_sound_path(TcoreHal *hal, int device)
+{
+	dbg("Entry");
+
+	return tcore_hal_set_sound_path(g_mux_obj_ptr->phy_hal, device);
+}
+
 /* CMUX supported HAL (Logical HAL) operations */
 static struct tcore_hal_operations mux_hops = {
 	.power = tcore_cmux_hal_power,
 	.send = tcore_cmux_hal_send,
+	.set_sound_path = tcore_cmux_hal_set_sound_path,
 };
 
 static TReturn tcore_cmux_send_data(int data_len, unsigned char *data)
-- 
1.7.10.4

