From a7806ad9a86716bcdc9acba58575015e6224af91 Mon Sep 17 00:00:00 2001
From: Guillaume Zajac <guillaume.zajac@linux.intel.com>
Date: Wed, 3 Oct 2012 10:16:53 +0200
Subject: [PATCH 03/10] hal: Add new method definitions
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 src/hal.c |   14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/src/hal.c b/src/hal.c
index 59ba41f..9ade963 100644
--- a/src/hal.c
+++ b/src/hal.c
@@ -241,6 +241,14 @@ TReturn tcore_hal_link_user_data(TcoreHal *hal, void *user_data)
 	return TCORE_RETURN_SUCCESS;
 }
 
+TReturn tcore_hal_link_object(TcoreHal *hal, CoreObject *co)
+{
+	if (!hal)
+		return TCORE_RETURN_EINVAL;
+
+	return hal->ops->link_object_channel(co);
+}
+
 void *tcore_hal_ref_user_data(TcoreHal *hal)
 {
 	if (!hal)
@@ -501,12 +509,14 @@ gboolean tcore_hal_get_power_state(TcoreHal *hal)
 	return hal->power_state;
 }
 
-TReturn tcore_hal_set_power(TcoreHal *hal, gboolean flag)
+TReturn tcore_hal_set_power(TcoreHal *hal, gboolean flag,
+				TcoreHalPowerCallBack func,
+				void *user_data)
 {
 	if (!hal || !hal->ops || !hal->ops->power)
 		return TCORE_RETURN_EINVAL;
 
-	return hal->ops->power(hal, flag);
+	return hal->ops->power(hal, flag, func, user_data);
 }
 
 TReturn tcore_hal_set_sound_path(TcoreHal *hal, int device)
-- 
1.7.10.4

