From 00565c321fc84175fbc3aa599b074d30f798e4ec Mon Sep 17 00:00:00 2001
From: Guillaume Zajac <guillaume.zajac@linux.intel.com>
Date: Mon, 8 Oct 2012 11:32:11 +0200
Subject: [PATCH 06/10] notification: Add new notifcation type
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 include/type/notification.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/include/type/notification.h b/include/type/notification.h
index 3fa9520..a196226 100644
--- a/include/type/notification.h
+++ b/include/type/notification.h
@@ -84,6 +84,7 @@ enum tcore_notification_command {
 	TNOTI_SS_RELEASE_COMPLETE,
 
 	TNOTI_MODEM = TCORE_NOTIFICATION | TCORE_TYPE_MODEM,
+	TNOTI_MODEM_ADDED,
 	TNOTI_MODEM_POWER,
 	TNOTI_MODEM_FLIGHT_MODE,
 	TNOTI_MODEM_DUN_PIN_CONTROL,
-- 
1.7.10.4

