From aef0af1166e2f7b6b3d53992774135891e379deb Mon Sep 17 00:00:00 2001
From: Guillaume Zajac <guillaume.zajac@linux.intel.com>
Date: Thu, 4 Oct 2012 15:00:41 +0200
Subject: [PATCH 08/10] hal: Add new setup_pdp entry point declaration and
 DATA mode
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 include/hal.h |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/include/hal.h b/include/hal.h
index bed6684..1e8a072 100644
--- a/include/hal.h
+++ b/include/hal.h
@@ -26,6 +26,7 @@ __BEGIN_DECLS
 typedef void (*TcoreHalReceiveCallback)(TcoreHal *hal, unsigned int data_len, const void *data, void *user_data);
 typedef enum tcore_hook_return (*TcoreHalSendHook)(TcoreHal *hal, unsigned int data_len, void *data, void *user_data);
 typedef void (*TcoreHalPowerCallBack)(TcoreHal *hal, void *user_data);
+typedef void (*TcoreHalSetupPDPCallBack)(CoreObject *co, const char *iname, void *user_data);
 
 enum tcore_hal_recv_data_type {
 	TCORE_HAL_RECV_INDICATION,
@@ -38,7 +39,8 @@ enum tcore_hal_mode {
 	TCORE_HAL_MODE_UNKNOWN,
 	TCORE_HAL_MODE_AT,
 	TCORE_HAL_MODE_CUSTOM,
-    TCORE_HAL_MODE_TRANSPARENT
+	TCORE_HAL_MODE_TRANSPARENT,
+	TCORE_HAL_MODE_DATA,
 };
 
 struct tcore_hal_operations {
@@ -47,6 +49,10 @@ struct tcore_hal_operations {
 	TReturn (*send)(TcoreHal *hal, unsigned int data_len, void *data);
 	TReturn (*set_sound_path)(TcoreHal *hal, int device);
 	TReturn (*link_object_channel)(CoreObject *object);
+	TReturn (*setup_pdp)(CoreObject *object,
+				TcoreHalSetupPDPCallBack func,
+				void *user_data,
+				unsigned int cid);
 };
 
 TcoreHal*    tcore_hal_new(TcorePlugin *plugin, const char *name,
@@ -68,7 +74,12 @@ TReturn	      tcore_hal_set_sound_path(TcoreHal *hal, int device);
 
 TReturn      tcore_hal_link_user_data(TcoreHal *hal, void *user_data);
 void*        tcore_hal_ref_user_data(TcoreHal *hal);
+
 TReturn	     tcore_hal_link_object(TcoreHal *hal, CoreObject *co);
+TReturn      tcore_hal_setup_pdp(TcoreHal *hal, CoreObject *co,
+					TcoreHalSetupPDPCallBack func,
+					void *user_data,
+					unsigned int cid);
 
 TReturn      tcore_hal_send_data(TcoreHal *hal, unsigned int data_len, void *data);
 TReturn      tcore_hal_send_request(TcoreHal *hal, TcorePending *pending);
-- 
1.7.10.4

