From ef579dee232a6fdddaf08c537c548cfa2e9d0240 Mon Sep 17 00:00:00 2001
From: Guillaume Zajac <guillaume.zajac@linux.intel.com>
Date: Thu, 4 Oct 2012 15:01:27 +0200
Subject: [PATCH 09/10] hal: Add tcore_hal_setup_pdp definition and mana DATA
 mode
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

---
 src/hal.c |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/hal.c b/src/hal.c
index 9ade963..579d765 100644
--- a/src/hal.c
+++ b/src/hal.c
@@ -249,6 +249,16 @@ TReturn tcore_hal_link_object(TcoreHal *hal, CoreObject *co)
 	return hal->ops->link_object_channel(co);
 }
 
+TReturn tcore_hal_setup_pdp(TcoreHal *hal, CoreObject *co,
+				TcoreHalSetupPDPCallBack func,
+				void *user_data,
+				unsigned int cid)
+{
+	if (!hal)
+		return TCORE_RETURN_EINVAL;
+
+	return hal->ops->setup_pdp(co, func, user_data, cid);
+}
 void *tcore_hal_ref_user_data(TcoreHal *hal)
 {
 	if (!hal)
@@ -356,6 +366,12 @@ TReturn tcore_hal_dispatch_response_data(TcoreHal *hal, int id,
 
 			/* Invoke CMUX receive API for decoding */
 			tcore_cmux_rcv_from_hal((unsigned char *)data, data_len);
+		} else if (hal->mode == TCORE_HAL_MODE_DATA) {
+			dbg("TCORE_HAL_MODE_DATA");
+
+			tcore_hal_emit_recv_callback(hal, data_len, data);
+
+			return TCORE_RETURN_SUCCESS;
 		}
 		/* Send next request in queue */
 		g_idle_add_full(IDLE_SEND_PRIORITY, _hal_idle_send, hal, NULL );
-- 
1.7.10.4

